### ISSUE TITLE
Bugs loading some Velox EMD files

### ISSUE THREAD

**Issue by din14970** [hyperspy/hyperspy#2483]
Of course just after I close my pull request on improving the Velox emd reader I find some datasets where the reader throws errors. The EMD files I am experiencing issues with can be downloaded here: https://owncloud.gwdg.de/index.php/s/D3oo5AEML7Zr9in

Datasets were collected in an image corrected Titan instrument with a Ceta camera. I'm using Hyperspy 1.6.0.

Dataset 2 has issues I believe because an FFT created in Velox may have been stored as an image dataset. The reader doesn't know how to handle this and I get this error:

```
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-33-188d3ff6bef4> in <module>
----> 1 hs.load(folder+datasets[2])

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io.py in load(filenames, signal_type, stack, stack_axis, new_axis_name, lazy, convert_units, escape_square_brackets, **kwds)
    337         else:
    338             # No stack, so simply we load all signals in all files separately
--> 339             objects = [load_single_file(filename, lazy=lazy,
    340                                         **kwds)
    341                        for filename in filenames]

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io.py in <listcomp>(.0)
    337         else:
    338             # No stack, so simply we load all signals in all files separately
--> 339             objects = [load_single_file(filename, lazy=lazy,
    340                                         **kwds)
    341                        for filename in filenames]

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io.py in load_single_file(filename, **kwds)
    381     else:
    382         reader = io_plugins[i]
--> 383         return load_with_reader(filename=filename, reader=reader, **kwds)
    384 
    385 

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io.py in load_with_reader(filename, reader, signal_type, convert_units, **kwds)
    387                      **kwds):
    388     lazy = kwds.get('lazy', False)
--> 389     file_data_list = reader.file_reader(filename,
    390                                         **kwds)
    391     objects = []

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io_plugins/emd.py in file_reader(filename, lazy, **kwds)
   1716             raise IOError("The file is not a supported EMD file.")
   1717     except Exception as e:
-> 1718         raise e
   1719     finally:
   1720         if not lazy:

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io_plugins/emd.py in file_reader(filename, lazy, **kwds)
   1698             _logger.debug('EMD file is a Velox variant.')
   1699             emd_reader = FeiEMDReader(lazy=lazy, **kwds)
-> 1700             emd_reader.read_file(file)
   1701         elif is_EMD_NCEM(file):
   1702             _logger.debug('EMD file is a Bekerley variant.')

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io_plugins/emd.py in read_file(self, f)
    962             self.p_grp = f.get('Presentation')
    963             self._parse_image_display()
--> 964         self._read_data(self.select_type)
    965 
    966     def _read_data(self, select_type):

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io_plugins/emd.py in _read_data(self, select_type)
    980         if self.im_type == 'Image':
    981             _logger.info('Reading the images.')
--> 982             self._read_images()
    983         elif self.im_type == 'Spectrum':
    984             self._read_single_spectrum()

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io_plugins/emd.py in _read_images(self)
   1071         # each of them
   1072         for image_sub_group_key in _get_keys_from_group(image_group):
-> 1073             image = self._read_image(image_group, image_sub_group_key)
   1074             if not self.load_images:
   1075                 # If we don't want to load the images, we stop here

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io_plugins/emd.py in _read_image(self, image_group, image_sub_group_key)
   1102             # much faster in case of chunked data.
   1103             data = np.empty(h5data.shape)
-> 1104             h5data.read_direct(data)
   1105             data = np.rollaxis(data, axis=2)
   1106 

/anaconda3/envs/temmeta/lib/python3.8/site-packages/h5py/_hl/dataset.py in read_direct(self, dest, source_sel, dest_sel)
    731 
    732             for mspace in dest_sel.broadcast(source_sel.mshape):
--> 733                 self.id.read(mspace, fspace, dest, dxpl=self._dxpl)
    734 
    735     def write_direct(self, source, source_sel=None, dest_sel=None):

h5py/_objects.pyx in h5py._objects.with_phil.wrapper()

h5py/_objects.pyx in h5py._objects.with_phil.wrapper()

h5py/h5d.pyx in h5py.h5d.DatasetID.read()

h5py/_proxy.pyx in h5py._proxy.dset_rw()

h5py/_proxy.pyx in h5py._proxy.H5PY_H5Dread()

OSError: Can't read data (no appropriate function for conversion path)
```

I suppose one could work around this by loading only the recognized datasets, but for users I think it would be most useful if datasets that can't be loaded for some reason are just not loaded and it is skipped but the user is warned about this.

Datasets 14 and 16 contain image series collected with the ceta camera, although I'm not sure why there are two datasets in each emd file. The error I get is: 

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-34-4774df661acd> in <module>
----> 1 hs.load(folder+datasets[14])

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io.py in load(filenames, signal_type, stack, stack_axis, new_axis_name, lazy, convert_units, escape_square_brackets, **kwds)
    337         else:
    338             # No stack, so simply we load all signals in all files separately
--> 339             objects = [load_single_file(filename, lazy=lazy,
    340                                         **kwds)
    341                        for filename in filenames]

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io.py in <listcomp>(.0)
    337         else:
    338             # No stack, so simply we load all signals in all files separately
--> 339             objects = [load_single_file(filename, lazy=lazy,
    340                                         **kwds)
    341                        for filename in filenames]

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io.py in load_single_file(filename, **kwds)
    381     else:
    382         reader = io_plugins[i]
--> 383         return load_with_reader(filename=filename, reader=reader, **kwds)
    384 
    385 

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io.py in load_with_reader(filename, reader, signal_type, convert_units, **kwds)
    387                      **kwds):
    388     lazy = kwds.get('lazy', False)
--> 389     file_data_list = reader.file_reader(filename,
    390                                         **kwds)
    391     objects = []

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io_plugins/emd.py in file_reader(filename, lazy, **kwds)
   1716             raise IOError("The file is not a supported EMD file.")
   1717     except Exception as e:
-> 1718         raise e
   1719     finally:
   1720         if not lazy:

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io_plugins/emd.py in file_reader(filename, lazy, **kwds)
   1698             _logger.debug('EMD file is a Velox variant.')
   1699             emd_reader = FeiEMDReader(lazy=lazy, **kwds)
-> 1700             emd_reader.read_file(file)
   1701         elif is_EMD_NCEM(file):
   1702             _logger.debug('EMD file is a Bekerley variant.')

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io_plugins/emd.py in read_file(self, f)
    962             self.p_grp = f.get('Presentation')
    963             self._parse_image_display()
--> 964         self._read_data(self.select_type)
    965 
    966     def _read_data(self, select_type):

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io_plugins/emd.py in _read_data(self, select_type)
    980         if self.im_type == 'Image':
    981             _logger.info('Reading the images.')
--> 982             self._read_images()
    983         elif self.im_type == 'Spectrum':
    984             self._read_single_spectrum()

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io_plugins/emd.py in _read_images(self)
   1071         # each of them
   1072         for image_sub_group_key in _get_keys_from_group(image_group):
-> 1073             image = self._read_image(image_group, image_sub_group_key)
   1074             if not self.load_images:
   1075                 # If we don't want to load the images, we stop here

/anaconda3/envs/temmeta/lib/python3.8/site-packages/hyperspy/io_plugins/emd.py in _read_image(self, image_group, image_sub_group_key)
   1122             i = 0
   1123         else:
-> 1124             frame_time = original_metadata['Scan']['FrameTime']
   1125             frame_time, time_unit = self._convert_scale_units(
   1126                 frame_time, 's', 2 * data.shape[0])

KeyError: 'FrameTime'
```
To me it looks like somewhere along the way it assumes we are dealing with a STEM dataset but it is TEM, which of course will not the right entries in the metadata. It's strange because a similar dataset, dataset 20 (https://owncloud.gwdg.de/index.php/s/LZ9VLB0Txb5YkVW) is very similar to these and this one could be loaded.

If I can find some time I'll have a look whether I could make a quick fix to these issues.

**Comment by ericpre** [comment]
If you have a chance, try to generate as small as possible file to add to the test suite.

**Comment by din14970** [comment]
Is someone working on this or should I give it a stab?

**Comment by ericpre** [comment]
Sounds like it all yours! ;)

**Comment by ericpre** [comment]
Fixed in #2564.
